### PR TITLE
Slas 201 nickname email

### PR DESCRIPTION
**Pull request by nedrocks**
image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@slashauth/slashauth-react",
-  "version": "0.6.6",
+  "version": "0.7.0-beta.0",
```
Comment (nsalhuan) [inline package.json]: nice!

File: src/api.ts [hunk 2]
```diff
@@ -68,14 +69,41 @@ export async function getNonceToSign({
...
+
 export async function loginWithSignedNonce({
   baseUrl,
   ...options
 }: LoginWithSignedNonceOptions) {
   const queryString = createQueryParams(options);
   return await getJSON<LoginWithSignedNonceResponse>(
     `${baseUrl}/loginWithSignedNonce?${queryString}`,
-    1000,
+    10000,
```
Comment (nsalhuan) [inline src/api.ts]: would three seconds be enough?

File: src/hooks/use-modal-core.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    setWagmiConnector(connector);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }
+
+  if (wagmiConnector && !connectModal) {
+    const modalCore = new ModalCore(wagmiConnector);
+    setConnectModal(modalCore);
+
+    // TODO: Hide the modal when everything is connected!
```
Comment (nsalhuan) [inline src/hooks/use-modal-core.ts]: still a todo?

File: src/hooks/use-modal-core.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  if (connectModal && !connectModal.appConfig && appConfig) {
+    connectModal.appConfig = appConfig;
+  }
+
+  const handleDeactivate = useCallback(() => {
+    wagmiConnector?.clearState();
+
+    // TODO: Disconnect listeners?
```
Comment (nsalhuan) [inline src/hooks/use-modal-core.ts]: ditto

File: src/modal/core.tsx [hunk 5]
```diff
@@ -34,6 +108,7 @@ export class ModalCore {
   }
 
   public async hideModal(): Promise<void> {
+    console.trace('hiding');
```
Comment (nsalhuan) [inline src/modal/core.tsx]: delete

File: src/provider/wallet-auth.tsx [hunk 6]
```diff
@@ -25,31 +29,45 @@ const initialState = {
...
+      eventEmitter.emit(ACCOUNT_CONNECTED_EVENT, {
+        account,
+      });
+    }
+    return account || null;
+  };
+
   const handleDeactivate = useCallback(() => {
-    wagmiConnector?.clearState();
+    // wagmiConnector?.clearState();
```
Comment (nsalhuan) [inline src/provider/wallet-auth.tsx]: delete?

**Review by nsalhuan — APPROVED** [review]

File: src/hooks/use-modal-core.ts (hunk 4 above)
Comment (nedrocks) [reply src/hooks/use-modal-core.ts]: Handled.

File: src/hooks/use-modal-core.ts (hunk 3 above)
Comment (nedrocks) [reply src/hooks/use-modal-core.ts]: Handled

File: src/api.ts (hunk 2 above)
Comment (nedrocks) [reply src/api.ts]: I figure let's go lnog for now -- we can start tracking analytics!

**Review by nsalhuan — APPROVED** [review]

**Comment by nedrocks** [comment]
Going to merge!

**Pull request merged**